src/main/java/com/arpnetworking/metrics/mad/model/json/Version2g.java

```diff
@@ -0,0 +1,394 @@
+            _counters = value;
+            return this;
+        }
+
+        /**
+         * Sets the timers field.
+         *
+         * @param value Value
+         * @return This builder
+         */
+        public Builder setTimers(final ImmutableMap<String, Element> value) {
+            _timers = value;
+            return this;
+        }
+
+        /**
+         * Sets the gauges field.
+         *
+         * @param value Value
+         * @return This builder
+         */
+        public Builder setGauges(final ImmutableMap<String, Element> value) {
+            _gauges = value;
+            return this;
+        }
+
+        @NotNull
+        private String _id;
+        @NotNull
+        private DateTime _start;
+        @NotNull
+        private DateTime _end;
+        @NotNull
+        private ImmutableMap<String, String> _dimensions;
+        @NotNull
+        private ImmutableMap<String, String> _annotations;
+        private ImmutableMap<String, Element> _counters;
+        private ImmutableMap<String, Element> _gauges;
+        private ImmutableMap<String, Element> _timers;
+        @NotNull
+        @MatchPattern(pattern = "^2g$", flags = Pattern.CASE_INSENSITIVE)
+        private String _version;
+    }
+
+    /**
+     * Represents a single sample.
+     */
+    public static final class Sample {
+
+        public Unit2g getUnit2g() {
+            return _unit2g;
+        }
+
+        public double getValue() {
+            return _value;
+        }
+
+        private Sample(final Builder builder) {
+            _unit2g = builder._unit2g;
+            _value = builder._value;
+        }
+
+        private final Unit2g _unit2g;
+        private final double _value;
+
+        /**
+         * Builder for the Sample class.
+         */
+        public static final class Builder extends OvalBuilder<Sample> {
+            /**
+             * Public constructor.
+             */
+            public Builder() {
+                super(Sample::new);
+            }
+
+            /**
+             * Sets the value field.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setValue(final Double value) {
+                _value = value;
+                return this;
+            }
+
+            /**
+             * Sets the compound unit field.
+             *
+             * @param value Value
+             * @return This builder
+             */
+            public Builder setUnit(final Unit2g value) {
+                _unit2g = value;
+                return this;
+            }
+
+            @NotNull
+            private Double _value;
+            private Unit2g _unit2g;
+        }
+
+        /**
+         * 2G format Unit class.
+         */
+        public static final class Unit2g {
+
+            public List<CompoundUnit> getNumerators() {
+                return _numerators;
```

Comment:
I am open to naming suggestions.  I initially used CompoundUnit because I was splitting the existing units ( millisecond -> { milli, second } )

Unit2g -> CompoundUnit (because it contains the numerator/denominator lists)
CompoundUnit -> Unit